src/bootstrap/builder.rs

```diff
@@ -466,11 +469,12 @@ impl<'a> ShouldRun<'a> {
         // exceptional case for `Kind::Setup` because its `library`
         // and `compiler` options would otherwise naively match with
         // `compiler` and `library` folders respectively.
-        assert!(
-            self.kind == Kind::Setup || !self.builder.src.join(alias).exists(),
-            "use `builder.path()` for real paths: {}",
-            alias
-        );
+        match self.kind == Kind::Setup || !self.builder.src.join(alias).exists() {
+            true => {}
+            false => {
+                eprintln!("use `builder.path()` for real paths: {:?}", alias)
+            }
+        };
```

Comment:
`if` statement would better fit here

Reply:
this is not a user-facing error message, this only happens if there's a bug in a Step. so I think an assert like before is better.